### DISCUSSION TITLE
Can You Edit Automatically Generated Release Workflow?

### DISCUSSION THREAD

**Discussion by hakenmt** [projen/projen#3790] [Q&A] [answered]
My workflow is running on a self-hosted runner using CodeBuild for an AWS CDK project. One of the automatically created workflows for `release` includes a `release-pypi` job. One of the steps in that automatically created jobs is `setup-python`, which fails since Amazon Linux isn't supported. You can't manually remove it locally since it's recreated during the build, is there a programmatic way in the projen.ts file to remove that step?

**Comment by mrgrain** [comment, marked as answer]
Yes you can, you need to use the [low-level file patching API](https://projen.io/docs/concepts/escape-hatches#object-file-patches). Something like this:

```ts
project.github?.tryFindWorkflow('release')?.file.patch(...)
```

**Comment by mrgrain** [comment]
But also, the release-pypi job is only added if you have configured to release to PyPI. I suggest you are not configuring a PyPI release and then you should not need to change the workflow at all.

**Reply by hakenmt** [reply]
Great, this worked and was definitely simpler than trying to manipulate the resulting `release.yml` directly.

```
project.github?.tryFindWorkflow('release')?.file?.patch(JsonPatch.remove('/jobs/release_pypi/steps/1'));
```

This removes the `setup-python` step so the rest of that job can run successfully in an AL2023 Codebuild environment.
